src/tools/miri/src/shims/x86/sse.rs

```diff
@@ -0,0 +1,466 @@
+
+                assert_eq!(left_len, right_len);
+
+                let left = this.read_scalar(&this.mplace_index(&left, 0)?.into())?.to_f32()?;
+                let right = this.read_scalar(&this.mplace_index(&right, 0)?.into())?.to_f32()?;
+                let res = match unprefixed_name {
+                    "comieq.ss" | "ucomieq.ss" => left == right,
+                    "comilt.ss" | "ucomilt.ss" => left < right,
+                    "comile.ss" | "ucomile.ss" => left <= right,
+                    "comigt.ss" | "ucomigt.ss" => left > right,
+                    "comige.ss" | "ucomige.ss" => left >= right,
+                    "comineq.ss" | "ucomineq.ss" => left != right,
+                    _ => unreachable!(),
+                };
+                this.write_scalar(Scalar::from_i32(i32::from(res)), dest)?;
+            }
+            "cvtss2si" | "cvttss2si" => {
+                let [op] = this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let (op, _) = this.operand_to_simd(op)?;
+
+                let op = this.read_scalar(&this.mplace_index(&op, 0)?.into())?.to_f32()?;
+
+                let rnd = match unprefixed_name {
+                    "cvtss2si" => rustc_apfloat::Round::NearestTiesToEven,
+                    "cvttss2si" => rustc_apfloat::Round::TowardZero,
+                    _ => unreachable!(),
+                };
+
+                let mut exact = false;
+                let cvt = op.to_i128_r(32, rnd, &mut exact);
+                let res = if cvt.status.intersects(
+                    rustc_apfloat::Status::INVALID_OP
+                        | rustc_apfloat::Status::OVERFLOW
+                        | rustc_apfloat::Status::UNDERFLOW,
+                ) {
+                    i32::MIN
+                } else {
+                    cvt.value as i32
+                };
+
+                this.write_scalar(Scalar::from_i32(res), dest)?;
+            }
+            "cvtss2si64" | "cvttss2si64" => {
+                let [op] = this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let (op, _) = this.operand_to_simd(op)?;
+
+                let op = this.read_scalar(&this.mplace_index(&op, 0)?.into())?.to_f32()?;
+
+                let rnd = match unprefixed_name {
+                    "cvtss2si64" => rustc_apfloat::Round::NearestTiesToEven,
+                    "cvttss2si64" => rustc_apfloat::Round::TowardZero,
+                    _ => unreachable!(),
+                };
+
+                let mut exact = false;
+                let cvt = op.to_i128_r(64, rnd, &mut exact);
+                let res = if cvt.status.intersects(
+                    rustc_apfloat::Status::INVALID_OP
+                        | rustc_apfloat::Status::OVERFLOW
+                        | rustc_apfloat::Status::UNDERFLOW,
+                ) {
+                    i64::MIN
+                } else {
+                    cvt.value as i64
```

Comment:
`as` casts are dangerous and if you had made this a Miri PR, CI would have failed due to a clippy check. We always use checked casts to make sure this doesn't overflow.